queries/go.scm

```diff
@@ -222,3 +222,55 @@
     .
   )
 ) @anonymousFunction @namedFunction
+
+;; switch-based branch
+
+(
+  [
+    (default_case)
+    (expression_case)
+    (type_case)
+  ] @branch
+  (#trim-end! @branch)
+  (#insertion-delimiter! @branch "\n")
+)
+
+[
+  (type_switch_statement)
+  (expression_switch_statement)
+] @branch.iteration
+
+;; if-else-based branch
+
+;; first if in an if-else chain
+(
+  (if_statement
+    consequence: (block) @branch.end.endOf
+  ) @_if @branch.start.startOf
+  (#not-parent-type? @_if if_statement)
+  (#insertion-delimiter! @branch.start.startOf " ")
+)
+
+;; internal if in an if-else chain
+(if_statement
+  "else" @branch.start.startOf
+  alternative: (if_statement
+    consequence: (block) @branch.end.endOf
```

Comment:
I'm pretty sure you don't need the `.startOf` / `.endOf` if they're not overlapping and you can rely on the order
```suggestion
  "else" @branch.start
  alternative: (if_statement
    consequence: (block) @branch.end
```